New Float Function

jestdang opened the pull request:
This pull requests adds the following features:

1. Added a new float function to generate random float values in RcppDeepState.h
2. Added a new datatype description column for the float function in IntegerMatrix

jestdang committed:
Added a function to randomly generate float values.

jestdang committed:
Added new row for float datatype in types_table.

FabrizioSandri commented:
@jestdang, that's fantastic! Thank you for including support for this new datatype.

FabrizioSandri committed:
missing closed parentheses

FabrizioSandri committed:
missing Maintainer in the description file

jestdang committed:
Added float in range function

RcppDeepState_float(float min, float max) function

jestdang committed:
Updated RcppDeepState_float() function

Changed RcppDeepState_float(float min, float max) to RcppDeepState_float(float low, float high)

jestdang committed:
Changed datatype for float function

changed NA, to (low, high)

jestdang committed:
updated datatype for float function

added quotes

FabrizioSandri commented on R/fun_harness_create.R:
```diff
@@ -40,7 +40,8 @@ deepstate_fun_create <- function(package_path, function_name, sep="infun"){
     datatype("CharacterVector", NA, NA),
-    datatype("mat", NA, NA))
+    datatype("mat", NA, NA),
+    datatype("float", "NumericVector", "(low,high)")
```
@jestdang you missed the close parentheses here.

jestdang committed:
added comma

updated datatype for float function

FabrizioSandri commented:
Great, CI tests passed and support for the `float` datatype has been added.
Thanks @jestdang
